### PR TITLE
GenerateChangelog task

### PR DESCRIPTION
**Pull request by aly76**
- Features a new task that can be added to release pipelines, and will generate a release changelog using the current release definition and the previous release definition stored in a json.
  - Uses Azure Release API to fetch the alias of artifacts in the release
  - CLI gets the release definition from an input artifact directory

- The changelog, in markdown and json format, will be pushed to a user-specified repository & branch

- Changes to ArtifactFileFetcher logic
  - Globs the artifact alias or pipeline workspace, depending on the artifact type

- Package changelog, consisting of commits up till artifact source version, is bundled with artifact
  - ArtifactFileFetcher returns the changelogFilepath

### CODE REVIEW

File: packages/azpipelines/BuildTasks/GenerateChangelogTask/task.json [hunk 1]
```diff
@@ -0,0 +1,149 @@
+{
+    "id": "87d5162d-07ee-40ea-8bf4-ac8dfdb907f1",
+    "name": "sfpowerscripts-generatechangelog-task",
+    "friendlyName": "Generate Changelog [ALPHA]",
+    "description": "Generates changelog for all the artifacts that is in a release. This task is part of sfpowerscripts, and requires SFDX plugin @dxatscale/sfpowerscripts@alpha to work",
+    "helpMarkDown": "SFDX plugin @dxatscale/sfpowerscripts@alpha needs to be installed in order for this task to function",
```
Comment (azlam-abdulsalam) [inline packages/azpipelines/BuildTasks/GenerateChangelogTask/task.json]: This is not needed any more

File: packages/azpipelines/BuildTasks/GenerateChangelogTask/task.json [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+          "type": "string",
+          "label": "Release Name",
+          "defaultValue": "",
+          "required": true,
+          "helpMarkDown": "Name of the release to be displayed in the changelog"
+        },
+        {
+          "name": "limit",
+          "type": "string",
+          "label": "Display Limit",
```
Comment (azlam-abdulsalam) [inline packages/azpipelines/BuildTasks/GenerateChangelogTask/task.json]: Some more verbosity

File: packages/azpipelines/vss-extension.json [hunk 3]
```diff
@@ -41,6 +41,7 @@
         }
     ],
     "scopes": [
+        "vso.code_write",
```
Comment (aly76) [inline packages/azpipelines/vss-extension.json]: Remove

**Review by azlam-abdulsalam — CHANGES_REQUESTED** [review]
Need a minor changes and we are good to merge

**Pull request merged**